sklearn/calibration.py

```diff
@@ -1099,6 +1099,9 @@ def plot(self, *, ax=None, name=None, ref_line=True, **kwargs):
             ax.plot([0, 1], [0, 1], "k:", label=ref_line_label)
         self.line_ = ax.plot(self.prob_pred, self.prob_true, "s-", **line_kwargs)[0]
 
+        # Unify AxesSubplot string representation for labels across matplotlib versions
+        self.line_.set_label(self.line_.get_label().replace("child", "line"))
```

Comment:
Shall we do that or only change the test. I am wondering what is the impact on the rendering of one of the figures.

Reply:
OK, so the plot did not change so I would only change the test. Indeed, the test is weird because we should only check that the line is not labelled in the legend instead of the current assert.

Reply:
So we should assert on `vis.ax_.get_legend()`.

In this regard, I think that we have an inconsistency. We should at least should always show the legend of the diagonal line named "Perfectly calibrated".

Reply:
I do not entirely see what you suggest, mainly because I do not know this part of the code base enough.

If you see it solved in a nicer way, feel free to push to this PR. :slightly_smiling_face:

Reply:
I will do